calico-cloud/reference/resources/runtimesecurity.mdx

```diff
@@ -0,0 +1,109 @@
+---
+description: API for this Calico Cloud resource 
+---
+
+# RuntimeSecurity
+
+The **RuntimeSecurity** custom resource (CR) is used to enable and configure Container Threat Detection in a Calico Cloud managed cluster.
+
+### Resource Definition
+
+```yaml
+apiVersion: operator.tigera.io/v1
+kind: RuntimeSecurity
+metadata:
+  name: defaultmy-runtimesecurity-config
+spec:
+  detectorConfig:
+    - id: 190
+      disabled: false
+    - id: 330
+      disbaled: True
+  runtimeExceptionList:
+    - matching: regex
+      processInvocation: "/bin/ls*"
+      pod: "not-evil-pod"
+      namespace: "default"
+    - matching: exact
+      pod: "nginx"
+      namespace: default
+    - matching:
+      namespace: "company-operations"
+```
+
+## Runtime Security Definition 
+
+### Metadata
+
+| Field  | Description                                   | Accepted Values                           | Schema |
+| ------ | --------------------------------------------- | ----------------------------------------- | ------ |
+| name   | The name of the runtime security resource.    | Lower-case alphanumeric with optional `-` | string |
+| labels | A set of labels to apply to this resource.    |                                           | map    |
+
+
+### Spec
+
+| Field                    | Description                                                                                  | Accepted Values  | Schema                                                | Default |
+| ------------------------ | -------------------------------------------------------------------------------------------- | ---------------- | ----------------------------------------------------- | ------- |
+| detectorConfig           | Configuration parameters for a detector in a Calico Cloud managed cluster.                   |                  | [DetectorConfig](#detectorconfig)                     |         |
+| runtimeExceptionList     | List of entries of suspicious processes that are allowed to run that wont generate an event  |                  | [runtimeExceptionList](#runtimeexceptionlist)         | Enabled |
+| runtimereporterdaemonset | Daemonset spec to override the default daemonset spec for runtime reporter                   |                  | [RuntimeReporterDaemonSet](#runtimereporterdaemonset) |         |          
```

Comment:
Oh dear, I hate this, and it's really problematic for documentation.  Firstly I don't think this is true, i.e. it can't be used to override the whole spec.  I believe we only use it for overriding resource limits and requests.  Secondly, I'm not sure we intend it for users at all, so I wonder if we should document the field at all.

Reply:
What about the RuntimeReporterConfig field?  This one is fully implemented, but problematic to document, unless we want to expose all that in our docs.

I think a reasonable solution would be to omit both RuntimeReporterConfig and runtimereporterdaemonset from this doc, and add a note somewhere saying "There are other fields in the RuntimeSecurity CR, but these are not yet designed for customer use".

Reply:
Yeah I purposely left out RuntimeReporterConfig, The only reason the RuntimeReporterDaemonset was left in was because other components that have this field document it as well. But I'm more than happy to remove it